### PR TITLE
fix bug where gts was displaying "IDLE STAFF" when all slots were occ…

### LINKED ISSUE #362: GTS falsely displays "IDLE STAFF" even though all staff slots are occupied
GTS_idle

### PR DESCRIPTION
**Pull request by statusNone**
closes #362

### CODE REVIEW

File: CovertInfiltration/Src/CovertInfiltration/Classes/X2Helper_Infiltration.uc [hunk 1]
```diff
@@ -539,6 +539,30 @@ static function XComGameState_MissionSite GetMissionStateFromActivity (XComGameS
 	return XComGameState_MissionSite(`XCOMHISTORY.GetGameStateForObjectID(ActivityState.PrimaryObjectRef.ObjectID));
 }
 
+static function XComGameState_HeadquartersProjectTrainAcademy GetTrainAcademyProject(StateObjectReference UnitRef)
```
Comment (Xymanek) [inline CovertInfiltration/Src/CovertInfiltration/Classes/X2Helper_Infiltration.uc]: Why was `X2Helper_Infiltration::GetAcademyProjectForUnit` not enough?

**Comment by Xymanek** [comment]
You should also replace `GTSTemplate.GetQueueMessageFn` since the vanilla one references the rookie training project

File: CovertInfiltration/Src/CovertInfiltration/Classes/X2Helper_Infiltration.uc (hunk 1 above)
Comment (statusNone) [reply CovertInfiltration/Src/CovertInfiltration/Classes/X2Helper_Infiltration.uc]: i had no idea that existed

File: CovertInfiltration/Src/CovertInfiltration/Classes/X2Helper_Infiltration_TemplateMod.uc [hunk 2]
```diff
@@ -902,6 +903,37 @@ static function bool IsAcademyProjectActive(StateObjectReference FacilityRef)
...
+			AcademyProject = class'X2Helper_Infiltration'.static.GetAcademyProjectForUnit(StaffSlot.GetAssignedStaffRef());
+			if (AcademyProject != none)
+			{
+				if (AcademyProject.GetCurrentNumHoursRemaining() < 0)
+					Message = class'UIUtilities_Text'.static.GetColoredText(class'UIFacility_Powercore'.default.m_strStalledResearch, eUIState_Warning);
+				else
+					Message = class'UIUtilities_Text'.static.GetTimeRemainingString(AcademyProject.GetCurrentNumHoursRemaining());
+
+				strStatus = StaffSlot.GetBonusDisplayString() $ ":" @ Message;
+				break;
```
Comment (Xymanek) [inline CovertInfiltration/Src/CovertInfiltration/Classes/X2Helper_Infiltration_TemplateMod.uc]: What happens if there are 2 people training currently?

Comment (Slivicon) [reply CovertInfiltration/Src/CovertInfiltration/Classes/X2Helper_Infiltration_TemplateMod.uc]: If it helps, I recently had 2 people training and still see the icon in the lower left about idle staff.

File: CovertInfiltration/Src/CovertInfiltration/Classes/X2Helper_Infiltration_TemplateMod.uc [hunk 3]
```diff
@@ -921,17 +922,24 @@ static function string GetAcademyQueueMessage(StateObjectReference FacilityRef)
...
+				iCurrentHoursRemaining = AcademyProject.GetCurrentNumHoursRemaining();
+				if (iCurrentHoursRemaining < 0)
+				{
 					Message = class'UIUtilities_Text'.static.GetColoredText(class'UIFacility_Powercore'.default.m_strStalledResearch, eUIState_Warning);
-				else
-					Message = class'UIUtilities_Text'.static.GetTimeRemainingString(AcademyProject.GetCurrentNumHoursRemaining());
-
-				strStatus = StaffSlot.GetBonusDisplayString() $ ":" @ Message;
-				break;
+					break;
```
Comment (Xymanek) [inline CovertInfiltration/Src/CovertInfiltration/Classes/X2Helper_Infiltration_TemplateMod.uc]: What about `strSoldierClass` in this case?

Comment (Xymanek) [reply CovertInfiltration/Src/CovertInfiltration/Classes/X2Helper_Infiltration_TemplateMod.uc]: Your change doesn't fix this problem....

**Pull request merged**